modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/FormBuilder/FormBuilder.es.js

```diff
@@ -12,8 +12,12 @@
  * details.
  */
 
-import {PagesVisitor, compose} from 'dynamic-data-mapping-form-renderer';
-import FormRenderer from 'dynamic-data-mapping-form-renderer/js/components/FormRenderer/FormRenderer.es';
+import {
+	FormNoopRenderer,
```

Comment:
Maybe we can find a better name for this. If I understand correctly, this is a `FormRenderer` that emits events but does not mutate the context when those events are fired, right? And `FormProvider` will act upon events like when `fieldRepeated` is fired it will actually insert a new field in the `DOM`. Is that it?

Reply:
> Maybe we can find a better name for this. If I understand correctly, this is a FormRenderer that emits events but does not mutate the context when those events are fired, right?

Yeah, I think it sounds like a good name due to the no-op and the Provider case just doing nothing.

> And FormProvider will act upon events like when fieldRepeated is fired it will actually insert a new field in the DOM. Is that it?

Yeah, there is a `useReducer` to control the local state about some changes that happen locally but not all events are processed by `FormProvider` some are done in `LayoutProvider`.

Reply:
> Yeah, I think it sounds like a good name due to the no-op and the Provider case just doing nothing.

to me it sounds like it renders nothing 🤣

Reply:
haha, I wasn't considering Renderer 😅, I was just taking into account the name of the FormRenderer component, not literally.